src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala

```diff
@@ -35,11 +35,14 @@ object XSDToSchema {
    * Reads a schema from an XSD file.
    *
    * @param xsdFile XSD file
+   * @param baseURI base URI for relative imports
    * @return Spark-compatible schema
    */
   @Experimental
-  def read(xsdFile: File): StructType = {
-    val xmlSchema = new XmlSchemaCollection().read(
+  def fromFile(xsdFile: File, baseURI: Option[String] = None): StructType = {
+    val xmlSchemaCollection = new XmlSchemaCollection()
+    xmlSchemaCollection.setBaseUri(baseURI.getOrElse(xsdFile.getParent))
```

Comment:
When would it not just be xsdFile.getParent?

Reply:
If the file you read is not at the root of your tree. E.g. if you decide to read a sub part of your tree that references something external

Reply:
I'd assume external resources have to be absolute anyway, but for the relative case, so you want to read `path/bar.xsd` from `foo.xsd` but `path/bar.xsd` doesn't exist at the relative path? that seems wrong. Is the case something like, you copied supporting XSDs to some location for deployment purposes that don't match how they're normally structured?

Reply:
I'm not saying this is the standard use case, hence the sane default.

Reply:
I'm not against supporting niche use cases; it's easy here. I'm just trying to understand how it arises at all

Reply:
I don't see any good reason to disallow this, do you want me to change something ? If not feel free to merge :)

Reply:
Eh, I think that's a little backwards - we'd add features because they solve a problem not because nobody can think of why not. I'm sure there is a use case for it, but, what is it? can you give any more real world example?

Reply:
This is not my use case, I agree to not make it complicated for ourselves. I will revert that, do you still want to keep the `.read` overloaded method then ? I'd rather not given the limitation mentioned in the other comment.

Reply:
OK, if we don't know a use for it, let's not do it yet. I'm open to the idea, just want to make sure it's not being added when there's a better/easier way.
I'm OK with renaming but is there then a reason left to do it? it's also solvable with another overload and no matter what there just won't be many overloads of this or args to methods anyway

Reply:
Only did minimal changes then